### PR TITLE
UX-1519/New menu element

### PR DESCRIPTION
**Pull request by jruddell**
image
image

### CODE REVIEW

File: src/app/core/elements/menu/menu-item.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  textVariant?: TextVariant
+  icon?: string
+  iconPlacement?: 'start' | 'end'
+  iconProps?: { [key: string]: any }
+  src?: string
+  className?: string
+  onClick?: () => void
+}
+
+export default function MenuItem({
```
Comment (patil-pratik-87) [inline src/app/core/elements/menu/menu-item.tsx:21]: 👌👌

File: src/app/core/elements/menu/index.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+export interface IMenuProps extends MenuPlacementProps {
+  open: boolean
+  anchor: JSX.Element
+  origin?: string
+  unorderedList?: boolean
+  className?: string
+  onClose?: () => void
+}
+
+export default function Menu({
```
Comment (patil-pratik-87) [inline src/app/core/elements/menu/index.tsx:19]: Nice 😍

File: src/app/core/elements/sidebar/index.tsx [hunk 3]
```diff
@@ -69,156 +70,179 @@ const Sidebar: FC<Props> = ({ setPluginId }) => {
...
-                </Text>
-              </Tooltip>
-            </li>,
-          ]}
+          className={classes.primaryPane}
+          bottomContent={currentSectionBottomLinks.map((bLink, idx) => (
+            <NavItem
+              {...bLink}
+              key={idx}
+              // className={singlePane ? classes.iconAndTextNavItem : undefined}
```
Comment (knnguy) [inline src/app/core/elements/sidebar/index.tsx:153]: comment

File: src/app/core/elements/sidebar/index.tsx [hunk 4]
```diff
@@ -69,156 +70,179 @@ const Sidebar: FC<Props> = ({ setPluginId }) => {
...
-                open
-                compact
-              />
-            )
-          })}
+          {currentSectionLinks?.map((navItem, idx) => (
+            <NavItem
+              {...navItem}
+              key={idx}
+              // className={singlePane ? classes.iconAndTextNavItem : classes.iconOnlyNavItem}
```
Comment (knnguy) [inline src/app/core/elements/sidebar/index.tsx:164]: comment

**Review by knnguy — APPROVED** [review]

File: src/app/core/elements/menu/index.tsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  origin = 'center center',
+  unorderedList = false,
+  onClose,
+  className,
+  children,
+  ...props
+}: PropsWithChildren<Props>) {
+  const stopClickRef = useRef(true)
+  const classes = useStyles({
+    vertAlign: align.verticle,
```
Comment (kevinylee1991) [inline src/app/core/elements/menu/index.tsx:36]: spelling nitpick: vertical -- this pops up 11 times though up to you if you want to fix

File: src/app/core/elements/sidebar/nav-item.tsx [hunk 6]
```diff
@@ -36,16 +36,10 @@ export default function NavItem({
   return (
     <Link to={link.path}>
       <li className={clsx(classes.navItem, className)}>
-        <Tooltip
-          message={tooltip ? name : ''}
-          className={classes.singlePaneNavText}
-          {...tooltipProps}
-        >
+        <Tooltip message={tooltip ? name : ''} {...tooltipProps}>
```
Comment (kevinylee1991) [inline src/app/core/elements/sidebar/nav-item.tsx:39]: is classes.singlePaneNavText not needed anymore? Not sure if still required for the user settings nav item when expanded

**Review by kevinylee1991 — APPROVED** [review]

File: src/app/core/elements/sidebar/nav-item.tsx (hunk 6 above)
Comment (jruddell) [reply src/app/core/elements/sidebar/nav-item.tsx:39]: not needed, I change that with the compact prop to the text element

**Pull request merged**